src/Sarif.Multitool/Rules/SARIF2010.ProvideCodeSnippets.cs

```diff
@@ -65,5 +78,46 @@ private void AnalyzeRegion(Region region, string regionPointer)
                     nameof(RuleResources.SARIF2010_ProvideCodeSnippets_Note_Default_Text));
             }
         }
+
+        private bool AnalyzeArtifactLocation(ArtifactLocation artifactLocation)
+        {
+            // no artifactLocation / no artifacts, so we should look for the snippet
+            if (artifactLocation == null || this.artifacts == null)
+            {
+                return true;
+            }
+
+            artifactLocation.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri resolvedUri);
+
+            foreach (Artifact artifact in this.artifacts)
+            {
+                // content/text doesn't exist, continue to next
+                if (string.IsNullOrEmpty(artifact.Contents?.Text))
+                {
+                    continue;
+                }
+
+                artifact.Location.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri artifactUri);
+
+                if (resolvedUri != null && artifactUri != null)
```

Comment:
>if (resolvedUri != null && artifactUri != null) [](start = 16, length = 47)

Since you will have previously checked the return values from the two calls to `TryReconstructAbsoluteUri`, you won't need this test.

Reply:
By the argument you made above, we _do_ need these tests.

---
In reply to: [462543518](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462543518) [](ancestors = 462543518)